generated/nitclk/session.py

```diff
@@ -0,0 +1,1140 @@
+
+    '''
+    return _get_session_class().configure_for_homogeneous_triggers(sessions)
+
+
+def finish_sync_pulse_sender_synchronize(sessions, min_time):
+    '''finish_sync_pulse_sender_synchronize
+
+    TBD
+
+    Args:
+        sessions (list of int): sessions is an array of sessions that are being synchronized.
+
+        min_time (float): Minimal period of TClk, expressed in seconds. Supported values are
+            between 0.0 s and 0.050 s (50 ms). Minimal period for a single
+            chassis/PC is 200 ns. If the specified value is less than 200 ns,
+            NI-TClk automatically coerces minTime to 200 ns. For multichassis
+            synchronization, adjust this value to account for propagation delays
+            through the various devices and cables.
+
+    '''
+    return _get_session_class().finish_sync_pulse_sender_synchronize(sessions, min_time)
+
+
+def init_for_documentation(self):
+    '''init_for_documentation
+
+    TBD
+    '''
+    return _get_session_class().init_for_documentation(self)
+
+
+def initiate(sessions):
+    '''initiate
+
+    Initiates the acquisition or generation sessions specified, taking into
+    consideration any special requirements needed for synchronization. For
+    example, the session exporting the TClk-synchronized start trigger is
+    not initiated until after initiate initiates all the sessions
+    that import the TClk-synchronized start trigger.
+
+    Args:
+        sessions (list of int): sessions is an array of sessions that are being synchronized.
+
+    '''
+    return _get_session_class().initiate(sessions)
+
+
+def is_done(sessions):
+    '''is_done
+
+    Monitors the progress of the acquisitions and/or generations
+    corresponding to sessions.
+
+    Args:
+        sessions (list of int): sessions is an array of sessions that are being synchronized.
+
+
+    Returns:
+        done (bool): Indicates that the operation is done. The operation is done when each
+            session has completed without any errors or when any one of the sessions
+            reports an error.
+
+    '''
+    return _get_session_class().is_done(sessions)
+
+
+def setup_for_sync_pulse_sender_synchronize(sessions, min_time):
+    '''setup_for_sync_pulse_sender_synchronize
+
+    TBD
+
+    Args:
+        sessions (list of int): sessions is an array of sessions that are being synchronized.
+
+        min_time (float): Minimal period of TClk, expressed in seconds. Supported values are
+            between 0.0 s and 0.050 s (50 ms). Minimal period for a single
+            chassis/PC is 200 ns. If the specified value is less than 200 ns,
+            NI-TClk automatically coerces minTime to 200 ns. For multichassis
+            synchronization, adjust this value to account for propagation delays
+            through the various devices and cables.
+
+    '''
+    return _get_session_class().setup_for_sync_pulse_sender_synchronize(sessions, min_time)
+
+
+def synchronize(sessions, min_time):
```

Comment:
Consider adding a default for `min_time` like the LabVIEW VIs do. The majority of users won't need to adjust this to anything greater than 200 ns.

Reply:
Also, `min_time` is a bit of an odd name for the parameter... `min_period` (or `min_tclk_period` if you're not into the whole brevity thing) would probably be more descriptive.

Reply:
Changed the name to `min_tclk_period`
Set default of 0
Allow datetime.timedelta() type